### ISSUE TITLE
Generic function `$` cannot be used as such

### ISSUE THREAD

**Issue by mjkallen** [bedatadriven/renjin#138]
Two examples: one where `obj` is a numeric vector and one where `obj` is a list. In both cases the method `$.foo` should return the string `"bar"`.

``` r
Renjin 0.8.1615
Copyright (C) 2015 The R Foundation for Statistical Computing
Copyright (C) 2015 BeDataDriven
> `$.foo` <- function(x,y) { "bar" }
> obj <- structure(1, class = "foo")
> obj$bar
ERROR: Invalid argument:
    $(double, symbol)
    Expected:
    $(Environment, any)
    $(list, any)
    $(PairList, any)
    $(ExternalPtr, any)
> obj <- structure(list(a=1), class = "foo")
> obj$bar
NULL
> 

```

**Comment by mjkallen** [comment]
`$` has been made generic in https://github.com/bedatadriven/renjin/commit/c92826fd7fb72842e1788e03fb9c817c02c73ed5, `$<-` remains to be done.

**Comment by akbertram** [comment]
Resolved by 055c0171843e31be022f07c8c7c72a87cf63afa9
